app/src/main/java/grinnell/appdev/edu/lyles/preferences/FavoritesManager.java

```diff
@@ -0,0 +1,108 @@
+package grinnell.appdev.edu.lyles.preferences;
+
+import android.content.Context;
+import static android.content.Context.MODE_PRIVATE;
+import android.content.SharedPreferences;
+
+import java.util.ArrayList;
+
+import grinnell.appdev.edu.lyles.MenuItem;
+
+/**
+ * A class which uses SharedPreferences to keep track of each user's favorite items in each category
+ *
+ * @author Shelby Frazier
+ */
+
+public class FavoritesManager {
+
+    private static final String PREFS_KEY = "Preferences";
+
+    private SharedPreferences mPreferencesFile;
+    private SharedPreferences.Editor mPreferencesEditor;
+    private ArrayList<MenuItem> mAllItemsArrayList;
+
+    /**
+     * Creates a new favorites manager based on an Arraylist of menuItems
+     *
+     * @param context   the context of activity using the FavoritesManager
+     * @param menuItems the list of all items the FavoritesManager will keep track of
+     */
+    public FavoritesManager(Context context, ArrayList<MenuItem> menuItems) {
+
+        mAllItemsArrayList = menuItems;
+        mPreferencesFile = context.getSharedPreferences(PREFS_KEY, MODE_PRIVATE);
+        mPreferencesEditor = mPreferencesFile.edit();
+
+        for(int i = 0; i < mAllItemsArrayList.size(); i++) {
+            String title = mAllItemsArrayList.get(i).getTitle();
+
+            if(!mPreferencesFile.contains(title)) {
+                mPreferencesEditor.putBoolean(title, false);
+                mPreferencesEditor.apply();
+            }
+        }
+    }
+
+    /**
+     * Changes whether the item with id id is a favorites
+     * This should be called by the click listener for any favorite button
+     *
+     * @param id the id of the item to be favorited or unfavorited
+     */
+    public void toggleFavorite(String id) {
+        boolean newValue = !mPreferencesFile.getBoolean(id, false);
+
+        mPreferencesEditor.remove(id);
```

Comment:
Can you check if this line is necessary? It seems odd to me that you'd have to remove and then re-put a value.